src/cpp/session/modules/SessionPath.cpp

```diff
@@ -39,129 +43,165 @@ namespace session {
 namespace modules { 
 namespace path {
 
-#ifdef __APPLE__
-
 namespace {
 
-Error readPathsFromFile(const FilePath& filePath,
-                        std::vector<std::string>* pPaths)
+bool containsPathEntry(
+      const std::vector<std::string>& pathEntries,
+      const std::string& entry)
 {
-   std::vector<std::string> paths;
-   Error error = core::readStringVectorFromFile(filePath, &paths);
-   if (error)
-   {
-      error.addProperty("path-source", filePath.getAbsolutePath());
-      return error;
-   }
-
-   std::copy(paths.begin(), paths.end(), std::back_inserter(*pPaths));
-
-   return Success();
+   // tolerate paths with trailing slashes
+   for (const std::string& item : { entry, entry + "/" })
+      if (core::algorithm::contains(pathEntries, item))
+         return true;
+   
+   return false;
+   
 }
 
-void safeReadPathsFromFile(const FilePath& filePath,
-                           std::vector<std::string>* pPaths)
+Error initializePathViaShell(const std::string& shellPath,
+                             std::string* pPath)
 {
-   Error error = readPathsFromFile(filePath, pPaths);
+   // double-check that the requested shell exists
+   if (!FilePath(shellPath).exists())
+      return fileNotFoundError(shellPath, ERROR_LOCATION);
+   
+   std::vector<std::string> args = { "-l", "-c", "printf \"%s\" \"$PATH\"" };
```

Comment:
what's -l and -c do here

Reply:
`-l` is used to ask the shell to behave like a login shell (so that it will source a user's shell profile, e.g. `.bash_profile` for bash).

`-c` is used to pass a command to the shell to be executed.

These flags are supported by the most commonly used macOS shells (bash, zsh, fish) so I think this invocation should be safe.